lib/core/src/sdk.rs

```diff
@@ -4031,6 +4050,115 @@ impl LiquidSdk {
         crate::logger::init_logging(log_dir, app_logger)
     }
+
+    pub async fn prepare_asset_swap(
+        &self,
+        req: &PrepareAssetSwapRequest,
+    ) -> Result<PrepareAssetSwapResponse, PaymentError> {
+        self.sideswap_service.clone().start().await?;
+
+        ensure_sdk!(
+            self.get_info().await?.wallet_info.balance_sat > req.send_amount_sat,
+            PaymentError::InsufficientFunds
+        );
+
+        let asset_id = AssetId::from_str(&req.asset_id)
+            .map_err(|err| PaymentError::generic(&format!("Invalid asset: {err}")))?;
+        let asset_swap = self
+            .sideswap_service
+            .subscribe_price_stream(asset_id, req.send_amount_sat)
+            .await?;
+        Ok(PrepareAssetSwapResponse { asset_swap })
+    }
+
+    pub async fn execute_asset_swap(
+        &self,
+        req: &ExecuteAssetSwapRequest,
+    ) -> Result<ExecuteAssetSwapResponse, PaymentError> {
+        self.sideswap_service.clone().start().await?;
+
+        let req_swap = &req.prepare_response.asset_swap;
+        let Some(ongoing_swap) = self.sideswap_service.get_current_price().await else {
+            return Err(PaymentError::generic(
+                "Cannot execute asset swap: swap not found.",
+            ));
+        };
+
+        ensure_sdk!(
+            ongoing_swap.fees_sat <= req_swap.fees_sat,
+            PaymentError::InvalidOrExpiredFees
+        );
+
+        ensure_sdk!(
+            req_swap.recv_amount == ongoing_swap.recv_amount
+                && req_swap.send_amount_sat == ongoing_swap.send_amount_sat,
+            PaymentError::generic(&format!(
+                "Asset swap amount mismatch: prev - send {} recv {} / new - send {} recv {}",
+                req_swap.send_amount_sat,
+                req_swap.recv_amount,
+                ongoing_swap.send_amount_sat,
+                ongoing_swap.recv_amount
+            ))
+        );
+
+        ensure_sdk!(
+            self.get_info().await?.wallet_info.balance_sat
+                >= ongoing_swap.fees_sat + req_swap.send_amount_sat,
+            PaymentError::InsufficientFunds
+        );
+
+        let ExecuteSwapResponse {
+            tx_id,
+            recv_address: destination,
+        } = self.sideswap_service.execute_swap().await?;
+        self.sideswap_service.stop().await;
+
+        // We insert a pseudo-tx in case LWK fails to pick up the new mempool tx for a while
+        // This makes the tx known to the SDK (get_info, list_payments) instantly
+        let tx_data = PaymentTxData {
+            tx_id: tx_id.clone(),
+            timestamp: Some(utils::now()),
+            amount: req_swap.recv_amount,
+            fees_sat: ongoing_swap.fees_sat,
+            payment_type: PaymentType::Send,
+            is_confirmed: false,
+            unblinding_data: None,
+            asset_id: req_swap.asset_id.clone(),
+        };
```

Comment:
Maybe I misunderstand, so if we're sending shouldn't the amount be the send_amount_sat and L-BTC asset? It's not really clear to me because I don't know what we'll see after the wallet scan